frontend/packages/dev-console/src/components/pipelines/modals/start-pipeline/StartPipelineModal.tsx

```diff
@@ -31,24 +35,45 @@ const StartPipelineModal: React.FC<StartPipelineModalProps & ModalComponentProps
       ...workspace,
       type: 'EmptyDirectory',
     })),
+    newSecrets: [],
     secretOpen: false,
+    secretForm: secretFormDefaultValues,
   };
 
-  const handleSubmit = (values: StartPipelineFormValues, actions): void => {
+  const handleSubmit = async (values: StartPipelineFormValues, actions) => {
     actions.setSubmitting(true);
+    let createSecrets: Promise<SecretKind>[] = [];
+    let secretFormError: boolean = false;
+    values.newSecrets.forEach((data) => {
+      const createSecret = k8sCreate(SecretModel, data);
+      createSecrets = [...createSecrets, createSecret];
+    });
 
-    submitStartPipeline(values, pipeline, userStartedLabel)
-      .then((res) => {
-        actions.setSubmitting(false);
-        onSubmit && onSubmit(res);
-        close();
-      })
-      .catch((err) => {
-        actions.setSubmitting(false);
-        actions.setStatus({ submitError: err.message });
-        errorModal({ error: err.message });
-        close();
-      });
+    if (!_.isEmpty(createSecrets)) {
+      await Promise.all(createSecrets)
```

Comment:
I think we need to take this secret logic into the `submitStartPipeline` function.

When we do that, we'll also want to make that function nicer. If we take the [resource logic](https://github.com/openshift/console/blob/master/frontend/packages/dev-console/src/components/pipelines/modals/start-pipeline/submit-utils.ts#L35-L63) and put it into it's own function that returns updated formValues. 

Then we can do the secret logic (probably just before that) and then everything is isolated and neat. Rather than odd logic bloat separated.

Ie, the logic flow of `submitStartPipeline` function could be:

```
const submitStartPipeline = (values, ...) => {
  // Isolated Secret logic
  await manageSecrets(...);

  // Move the resource logic to a new function
  const newValues = await managePipelineResources(values, ...);

  // Create Pipeline Run
  const pipelineRunResource: PipelineRun = await k8sCreate(
    PipelineRunModel,
    getPipelineRunFromForm(pipeline, formValues, labels),
  );

  return Promise.resolve(pipelineRunResource);
};
```
Note: Comments above are not needed, just making them for illustration point of view.